bmx280.c

```diff
@@ -309,6 +397,7 @@ static esp_err_t bmx280_calibrate(bmx280_t *bmx280)
     return ESP_OK;
 }
 
+#if !(CONFIG_USE_I2C_NEW_DEVICE)
```

Comment:
This macro sequence is a bit wack, I think you should probably split this into two like I suggest in the following review comments.